src/ccpp_constituent_prop_mod.F90

```diff
@@ -0,0 +1,203 @@
+module ccpp_constituent_prop_mod
+
+   ! ccpp_contituent_prop_mod contains types and procedures for storing
+   ! and retrieving constituent properties
+
+   implicit none
+   private
+
+   integer, parameter :: unassigned = -1
+
+   type, public :: ccpp_constituent_properties_type
+      character(len=:), private, allocatable :: std_name
+      character(len=:), private, allocatable :: vert_dim
+      integer,          private              :: field_ind = unassigned
+      logical,          private              :: advected = .false.
+   contains
+      ! Informational methods
+      procedure :: is_initialized     => ccp_is_initialized
+      procedure :: stdandard_name     => ccp_get_standard_name
+      procedure :: vertical_dimension => ccp_get_vertical_dimension
+      procedure :: field_index        => ccp_field_index
+      procedure :: is_advected        => ccp_is_advected
+      ! Methods that change state
+      procedure :: initialize         => ccp_initialize
+      procedure :: deallocate         => ccp_deallocate
+      procedure :: set_field_index    => ccp_set_field_index
+   end type ccpp_constituent_properties_type
+
+   private unassigned
+   private handle_allocate_error
+
+CONTAINS
+
+   subroutine handle_allocate_error(astat, fieldname, errflg, errmsg)
+      ! Generate an error message if <astat> indicates an allocation failure
+      integer,          intent(in)  :: astat
+      character(len=*), intent(in)  :: fieldname
+      integer,          intent(out) :: errflg
+      character(len=*), intent(out) :: errmsg
+
+      if (astat /= 0) then
+         errflg = astat
+         write(errmsg, '(4a,i0)') 'Error allocating ',                        &
+              'ccpp_constituent_properties_type object component, ',          &
+              trim(fieldname), ', error code = ', astat
+      else
+         errflg = 0
+         errmsg = ''
+      end if
+
+   end subroutine handle_allocate_error
+
+   logical function ccp_is_initialized(this, errflg, errmsg)
```

Comment:
What does ccp stand for? `ccpp_constituents_property` ?

Reply:
Correct. This is an internal name and when I originally wrote it (and the others) as ccpp, I was getting confused.
Ideally, users never need to look at this code. It is used by the suite-cap code that collects a suite's constituent fields and by some interfaces in the host cap that provide constituent services to host model.

Reply:
Thanks!